lib/smack/SmackInstGenerator.cpp

```diff
@@ -904,7 +904,7 @@ void SmackInstGenerator::visitIntrinsicInst(llvm::IntrinsicInst& ii) {
       body = Expr::bvExtract(var, width/2+1, width/2);
     }
     // Swap the bits to the right and the left of the already swapped portion.
-    unsigned offset = width%2==0 ? 0 : 1;
+    unsigned offset = width&1;
     for (unsigned i = width%2==0 ? 1 : 0; i < width/2; ++i) {
```

Comment:
Isn't `width%2==0 ? 1 : 0` equivalent to `offset^1`?

Reply:
Yes, but I want to keep the emphasis on its relation to the bit-width.

Reply:
`width^1` would also work, but `width%2==0 ...` is more obvious to me because of the parity.